proposals/0000-package-dependency-import-syntax.md

```diff
@@ -0,0 +1,159 @@
+
+*Revision*
+```swift
+@package(url: "http://github.com/author/example.git", .revision( "065675b3d1364a6f63b94a9c89be2e9ed0a4c3a1")) import Example
+```
+
+*Branch*
+```swift
+@package(url: "http://github.com/author/example.git", .branch("releases/1.0")) import Example
+```
+
+*Exact Version*
+```swift
+@package(url: "http://github.com/author/example.git", .exact("1.0")) import Example
+```
+
+*Range of version*
+```swift
+@package(url: "http://github.com/author/example.git", .from("1.0")) import Example
+```
+
+*Local dependency*
+Absolute path
+```swift
+@package(path: "/Users/author/example") import Example
+```
+
+Relative path
+```swift
+@package(path: "my_project/internal_package") import Example
+```
+
+#### Transitive dependencies 
+
+Package dependencies automatically make their transitive package dependencies available for use.
+
+For an example, lets look at the Package description below for our example dependency and transitive dependency.
+
+Direct dependency
+```swift
+let package = Package(
+    name: "Example",
+    products: [
+        .library(name: "example", targets: ["Example"]),
+    ],
+    dependencies: [
+        .package(url: "https://github.com/another_author/TransitiveDependency.git", from: "1.0.0"),
+    ],
+)
+```
+
+Transitive dependency
+```swift
+let package = Package(
+    name: "TransitiveDependency",
+    products: [
+        .library(name: "TransitiveDependency", targets: ["TransitiveDependency"]),
+    ],
+    dependencies: [],
+)
+```
+
+With the above Package descriptions, we can import transitive dependencies by explicitly exporting them in the package attribute definition.
+```swift
+import Foundation
+import UIKit
+@package(url: "http://github.com/author/example", products: ["TransitiveDependency"]) import Example 
+import TransitiveDependency 
+```
+
+### Integration with Swift Compiler 
+
+The `@package()` syntax will need to be added to the AST definition and compiler front-end will require changes to parse this syntax.
+
+### Integration with SwiftPM
+
+#### Execution
+Scripts that utilize the `@package()` attribute will be executed by `swift run` and managed by Swift package manager.
+
+Swift package manager will utilize the changes to the compiler front-end to access packages referenced using the `@package` attribute.
+
+#### Package resolution 
+Once package dependencies are known, SwiftPM will use it's existing logic for resolving those dependencies and outputting a ".resolved" file to ensure future invocations are reproducible.
+
+The name of the resolved file will be `script_name.resolved`.
+
+### Integration with SourceKit-LSP 
+
+TBD
+
+
+## Security
+
+TBD
+
+## Impact on exisiting packages
+
+This change will not affect existing packages
+
+
+## Alternatives considered
+
+TBD
```

Comment:
One alternative could be adding a feature to globally install packages that can be referenced by any script but that's usually bad because it makes the scripts non-portable.